Yoast/wordpress-seo
# Refactor social image preview

## Please test with
- JavaScript PR: https://github.com/Yoast/javascript/pull/652
- Wordpress-seo Premium PR: https://github.com/Yoast/wordpress-seo-premium/pull/2757

## Context

* The new social preview form should be connected to the store so that we can actually use it.

## Summary


This PR can be summarized in the following changelog entry:

* Connect to FacebookEditor and TwitterEditor to the redux store.

## Relevant technical choices:
* Redux has been moved from the monorepo to wordpress-seo. This keeps the component "dumb" and makes it easier to implement them in other places.
* The Selectors, Actions and Reducers have been split in two. The reason was that some duplication > less obvious code. Omar has approved this.
* I have chosen for a pragmatic way to determine post/taxonomy. This is something that definitely can be approved.
* Taxonomies seem to work. This means that https://github.com/Yoast/wordpress-seo/issues/14608 possibly can be closed 🎉  
* I have not tested with Premium.

## Test instructions

This PR can be tested by following these steps:
*Please test with:*
- JavaScript PR: https://github.com/Yoast/javascript/pull/652
- Wordpress-seo Premium PR: https://github.com/Yoast/wordpress-seo-premium/pull/2757
* Enter data in the Social Preview form and check that it is written to the hidden inputs (easy way to check is to see if the data persists after a reload when you update the page) and to the store (use redux dev tools).
* Test this for both taxonomies and post types.
* I have not tested with Premium, please do a quick check to see if it works and create an issue if it doesn't (I do not have a premium install running and I do not want to block this PR any longer).

## UI changes

## Documentation

## Quality assurance


Fixes #14910
